game/src/main/java/net/demilich/metastone/game/spells/AddEnchantmentSpell.java

```diff
@@ -110,8 +112,25 @@ protected void onCast(GameContext context, Player player, SpellDesc desc, Entity
 		if (enchantmentCard != null) {
 			List<Enchantment> enchantmentList = enchantmentCard.createEnchantments();
 			for (Enchantment enchantment : enchantmentList) {
-				enchantment.setOwner(player.getId());
-				context.getLogic().addGameEventListener(player, enchantment, target);
+				boolean yup = true;
```

Comment:
The effects you're trying to do might be better done with a `HasEnchantmentCondition` and an enchantment that lives on the player.